infrastructure/running-a-node/setup-full-node.md

```diff
@@ -1,4 +1,258 @@
+  
+    ```bash
+    cargo install --git https://github.com/paritytech/polkadot-sdk --tag <version> polkadot --locked
+    ```
+
+=== "Windows"
+    - Install [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
+    - Install [Ubuntu](https://docs.microsoft.com/en-us/windows/wsl/install-win10) (same webpage)
+    - Determine the latest version of the [Polkadot binary](https://github.com/paritytech/polkadot-sdk/tree/master/polkadot/releases)
+    - Download the correct Polkadot binary within Ubuntu by running the following command. Replace `*VERSION*` with the tag of the latest version from the last step (for example, `v0.8.22`):
+    ```bash
+    curl -sL https://github.com/paritytech/polkadot-sdk/releases/download/*VERSION*/polkadot -o polkadot
+    ```
+
+    - Then, run the following:
+    ```bash
+    sudo chmod +x polkadot
+    ```
+
+    - Which can be run as follows:
+    ```bash
+    ./polkadot --name "Your Node's Name"
+    ```
+
+    If Rust is installed, then `cargo` can be used similarly to macOS. The instructions for Linux may also be used.
+
+=== "Linux (Standalone)"
+    ??? info "`cannot execute binary file: Exec format error`"
+        The nature of pre-built binaries means that they may not work on your particular architecture or Linux distribution. If you see an error like `cannot execute binary file: Exec format error`, it likely means the binary is incompatible with your system. You will either need to compile the source code or use Docker.
+
+    - Determine the latest version of the [Polkadot binary](https://github.com/paritytech/polkadot-sdk/releases)
+
+    - Download the correct Polkadot binary within Ubuntu by running the following command. Replace `*VERSION*` with the tag of the latest version from the last step (e.g. `v0.8.22`):
+    ```bash
+    curl -sL https://github.com/paritytech/polkadot-sdk/releases/download/*VERSION*/polkadot -o polkadot
+    ```
+
+    - Then, run the following command: 
+    ```bash
+    `sudo chmod +x polkadot`
+    ```
+
+    - Which can be run as follows:
+    ```bash
+    polkadot --name "Your Node's Name"
+    ```
+
+=== "Linux (package)"
+
+    You can also install Polkadot from one of our package repositories. Installation from the Debian or rpm repositories will create a `systemd` service that can be used to run a Polkadot node. The service is inactive by default and will run as the `polkadot` user when started. Follow these steps to start the service and set it to auto-start on reboot:
+    
+    1. To start the service, run the following command:
+    ```bash
+    systemctl start polkadot
+    ```
+    2. Run this command to set the service to auto-start on reboot:
+    ```bash
+    systemctl enable polkadot
+    ```
+    
+    Command-line flags passed to the binary can be customized by editing `/etc/default/polkadot`. This file will not be overwritten on updating `polkadot`.
+
+### Debian-based (Debian, Ubuntu)
```

Comment:
This is confusing - why are there steps for each of the operating systems above and then these sections as well?